clowder-framework/clowder
# Expose extractor list to authenticated users

## Description
Demonstration: http://recordit.co/85PyXQxZoA

### Problem
The list of available extractors is currently only available to admins. It would be helpful to expose this list to non-admin users as well. This would help all users discover new extractors that may already be available for them to utilize. 

### Approach
* Move the "Extractors" link from the admin navbar sub-menu to "Explore > Extractors"
* Lock down the "Extractors" view so that it is read-only for non-admin users

### How to Test
Prerequisites: 2 users (1 admin + 1 non-admin)

1. Checkout and run this branch locally
2. Sign-up a non-admin user (if you haven't already) and login to Clowder as non-admin
3. From the navbar, choose "Explore > Extractors"
    * You should be brought to a read-only version of the enable/disable extractors list
    * You should be able to see (but not modify) which extractors are currently enabled on Clowder
    * You should **not** see any button beneath the list of extractors
   * Each extractor's name should appear as a link
4. Click an extractor's name
    * You should be brought to a read-only version of the ExtractorDetails view
    * You should be able to see (but not modify) details from the ExtractorInfo of the chosen extractor
    * You should **not** see a "Delete" button beneath the extractor info
5. Logout and log back into Clowder as an admin
6. From the navbar, choose "Explore > Extractors"
    * You should be brought to the writable version of the enable/disable extractors list which preserves existing admin functionality
    * You should be able to check/uncheck which extractors are currently enabled on Clowder
    * You should see Save/Cancel buttons beneath the list of extractors
   * Each extractor's name should appear as a link
7. Click an extractor's name
    * You should be brought to the writable version of the ExtractorDetails view which preserves existing admin functionality
    * You should be able to see (but not modify) details from the ExtractorInfo of the chosen extractor
    * You should see a "Delete" button beneath the extractor info

## Review Time Estimate


## Types of changes


## Checklist:
